### PR TITLE
Make allocator check C10_UNLIKELY

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78801

This popped up while having a look at posible causes for https://github.com/pytorch/pytorch/issues/78800

### CODE REVIEW

File: c10/core/impl/alloc_cpu.cpp [hunk 1]
```diff
@@ -58,27 +58,29 @@ void* alloc_cpu(size_t nbytes) {
   void* data;
 #ifdef __ANDROID__
   data = memalign(gAlignment, nbytes);
+  CAFFE_ENFORCE(
+      data,
+      "DefaultCPUAllocator: not enough memory: you tried to allocate ",
+      nbytes,
+      " bytes.");
```
Comment (Lezcano) [inline c10/core/impl/alloc_cpu.cpp:65]: These checks are moved inside the if-elifs blocks as the `else` branch already assures that `datta != nullptr` if `err == 0`.

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
Lint

**Comment by suo** [comment]
You are based on a super old master, and so the CI is not running some jobs which the mergebot requires

**Pull request closed without merging**